vsintegration/src/FSharp.Editor/CodeFix/CodeFixHelpers.fs

```diff
@@ -3,14 +3,33 @@
 namespace Microsoft.VisualStudio.FSharp.Editor
 
 open System.Threading
+open System.Threading.Tasks
+open System.Collections.Immutable
 
+open Microsoft
+open Microsoft.CodeAnalysis
 open Microsoft.CodeAnalysis.Text
 open Microsoft.CodeAnalysis.CodeFixes
 open Microsoft.CodeAnalysis.CodeActions
 open Microsoft.VisualStudio.FSharp.Editor.Telemetry
 
 [<RequireQualifiedAccess>]
 module internal CodeFixHelpers =
+    let reportCodeFixRecommendation (diagnostics: ImmutableArray<Diagnostic>) (doc: Document) (staticName: string) =
+        let ids =
+            diagnostics |> Seq.map (fun d -> d.Id) |> Seq.distinct |> String.concat ","
+
+        let props: (string * obj) list =
+            [
+                "name", staticName
+                "ids", ids
+                "context.document.project.id", doc.Project.Id.Id.ToString()
```

Comment:
Since bulk codefixes seem to work in a different way, we need to somehow mark in telemetry. I suggest adding a property "type"/"kind" and fill it with something like "bulk"/"batch"/"fixall", whatever is the most precise terminology.

Reply:
Will do that, yes. Good idea.